### ISSUE TITLE
Exception when dealing with delay signed assembly

### ISSUE THREAD

**Issue by lennylxx** [obfuscar/obfuscar#164]
I have the similar issue with this one https://github.com/obfuscar/obfuscar/issues/65.

**Repro step:**
1. Generate a key-pair `sn -k keypair.snk`
2. Get public key from the keypair `sn -p keypair.snk public.snk`
3. New a dot net core class library project, say `FooBar`, use below content:

**FooBar.csproj**
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <DelaySign>true</DelaySign>
    <SignAssembly>true</SignAssembly>
    <AssemblyOriginatorKeyFile>public.snk</AssemblyOriginatorKeyFile>
  </PropertyGroup>
  
  <ItemGroup>
      <PackageReference Include="Obfuscar" Version="2.2.13" />
  </ItemGroup>

  <Target Name="Obfuscating" AfterTargets="PostBuildEvent">
    <Exec Command=" &#34;$(Obfuscar)&#34; obfuscar.xml " ConsoleToMsBuild="true" />
  </Target>
</Project>

```
**obfuscar.xml**
```
<?xml version='1.0'?>
<Obfuscator>
  <Var name="InPath" value=".\bin\Debug\netstandard2.0" />
  <Var name="OutPath" value=".\Obfuscator_Output" />
  <Var name="KeyFile" value=".\public.snk" />

  <Module file="$(InPath)\FooBar.dll" />
</Obfuscator>
```
4. Build the project `dotnet build`

**Exception:**
```
PS C:\GitHub\FooBar> dotnet build
Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 32.66 ms for C:\GitHub\FooBar\FooBar.csproj.
  FooBar -> C:\GitHub\FooBar\bin\Debug\netstandard2.0\FooBar.dll
  '﻿setlocal' is not recognized as an internal or external command,
  operable program or batch file.
  Note that Rollbar API is enabled by default to collect crashes. If you want to opt out, please run with -s switch
  Loading project obfuscar.xml...Processing assembly: FooBar, Version=1.0.0.0, Culture=neutral, PublicKeyToken=ff34d3a52ba5d94f
  Loading assemblies...Extra framework folders: Done.
  hiding strings...
  Renaming:  fields...parameters...properties...events...methods...types...Done.

  Unhandled Exception: System.ArgumentException: Unable to obtain public key for StrongNameKeyPair.
     at System.Reflection.StrongNameKeyPair.ComputePublicKey()
     at System.Reflection.StrongNameKeyPair.get_PublicKey()
     at Mono.Cecil.ModuleWriter.Write(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
     at Mono.Cecil.ModuleWriter.WriteModule(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
     at Obfuscar.Obfuscator.SaveAssemblies(Boolean throwException)
     at Obfuscar.Obfuscator.RunRules()
     at Obfuscar.Program.Main(String[] args)
```
```
c:\GitHub\FooBar>sn -v bin\Debug\netstandard2.0\FooBar.dll

Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

bin\Debug\netstandard2.0\FooBar.dll is a delay-signed or test-signed assembly
```

The reason why I'm using delay sign is that my organization requires all assemblies should be signed on a dedicated signing server. And the `AssemblyOriginatorKeyFile` is a public key of certain keypair which is controlled by the signing server.
https://docs.microsoft.com/en-us/dotnet/framework/app-domains/delay-sign-assembly

**Comment by lextm** [comment]
I didn't investigate how Mono.Cecil can be used to delay sign an assembly. Pull requests are welcome.

The workaround is to use Obfuscar as the final step (after final signing the assemblies).

**Comment by lennylxx** [comment]
I tested the workaround you mentioned to obfuscate the final signed assembly, but still failed. Since I only have access to the public key, both assigning value and leaving blank for KeyFile will result exception.
Maybe the only way for delay sign is to add public key right after the obfuscating and before the final signing.

**Comment by lextm** [comment]
There seems to be a way to delay sign via Cecil. It just takes some time to write the actual code. You have to wait.

**Comment by lextm** [comment]
I just published 2.2.14 with experimental delay signing support for this issue. Can you test that version?

**Comment by lennylxx** [comment]
Thank you for your swift response, appreciate your work!
Confirmed 2.2.14 works.
